runtime/src/accounts_db.rs

```diff
@@ -5653,84 +5653,79 @@ pub mod tests {
 
-        const SLOT: Slot = 1;
-
-        let raw_expected = vec![
+        let mut raw_expected = vec![
             CalculateHashIntermediate {
                 version: 0,
-                hash: Hash::from_str("5K3NW73xFHwgTWVe4LyCg4QfQda8f88uZj2ypDx2kmmH").unwrap(),
+                hash: Hash::default(),
                 lamports: 1,
-                slot: SLOT,
+                slot,
                 pubkey: pubkey0,
             },
             CalculateHashIntermediate {
                 version: 1,
-                hash: Hash::from_str("84ozw83MZ8oeSF4hRAg7SeW1Tqs9LMXagX1BrDRjtZEx").unwrap(),
+                hash: Hash::default(),
                 lamports: 128,
-                slot: SLOT,
+                slot,
                 pubkey: pubkey127,
             },
             CalculateHashIntermediate {
                 version: 2,
-                hash: Hash::from_str("5XqtnEJ41CG2JWNp7MAg9nxkRUAnyjLxfsKsdrLxQUbC").unwrap(),
+                hash: Hash::default(),
                 lamports: 129,
-                slot: SLOT,
+                slot,
                 pubkey: pubkey128,
             },
             CalculateHashIntermediate {
                 version: 3,
-                hash: Hash::from_str("DpvwJcznzwULYh19Zu5CuAA4AT6WTBe4H6n15prATmqj").unwrap(),
+                hash: Hash::default(),
                 lamports: 256,
-                slot: SLOT,
+                slot,
                 pubkey: pubkey255,
             },
         ];
 
-        accounts.store_uncached(
-            SLOT,
-            &[(
-                &pubkey0,
-                &AccountSharedData::new(
+        let raw_accounts = vec![
+            AccountSharedData::new(
                     raw_expected[0].lamports,
                     1,
                     AccountSharedData::default().owner(),
                 ),
-            )],
-        );
-        accounts.store_uncached(
-            SLOT,
-            &[(
-                &pubkey127,
-                &AccountSharedData::new(128, 1, AccountSharedData::default().owner()),
-            )],
-        );
-        accounts.store_uncached(
-            SLOT,
-            &[(
-                &pubkey128,
-                &AccountSharedData::new(129, 1, AccountSharedData::default().owner()),
-            )],
-        );
-        accounts.store_uncached(
-            SLOT,
-            &[(
-                &pubkey255,
-                &AccountSharedData::new(256, 1, AccountSharedData::default().owner()),
-            )],
-        );
-        accounts.add_root(SLOT);
+            AccountSharedData::new(128, 1, AccountSharedData::default().owner()),
+            AccountSharedData::new(129, 1, AccountSharedData::default().owner()),
+            AccountSharedData::new(256, 1, AccountSharedData::default().owner()),
+        ];
+
+        for i in 0..raw_expected.len() {
+            raw_expected[i].hash =
+                AccountsDb::hash_account(slot, &raw_accounts[i], &raw_expected[i].pubkey);
```

Comment:
Just to confirm, do you have any concerns about replacing the hardcoded Hash values in the tests here?